test/common-algorithms/sorting/parallel/parallel_mergesort.chpl

```diff
@@ -0,0 +1,102 @@
+
+//merges two subarrays
+//First subarray is arr[l..m]
+//Second subarray is arr[m+1..r]
+
+proc merge(arr, left:int, mid:int, right:int){
+	var i: int = 0, j: int = 0, k: int = 0;
+	var left_length: int = mid - left + 1;
+	var right_length: int= right - mid;
+	
+	/* create temp arrays */
+	var left_arr: [1..left_length] int;	
+	var right_arr: [1..right_length] int;	
+			
+	/* Copy data to temp arrays L[] and R[] */
+	for i in (1..left_length){
+		//writeln("j inside = ", j);		
+		left_arr(i) = arr(left -1 + i);
+	}
+	
+	for j in (1..right_length){
+		//writeln("j inside = ", j);		
+		right_arr(j) = arr(mid + j);
+	}
+
+	i = 1;
+	j = 1;
+	k = left;
+	while (i <= left_length && j <= right_length)
+    	{
+		//writeln("i = ", i, "j = ", j, "k = ", k);		
+		if (left_arr(i) <= right_arr(j))
+		{
+		    arr(k) = left_arr(i);
+		    i+=1;
+		}
+		else
+		{
+		    arr(k) = right_arr(j);
+		    j+=1;
+		}
+		k+=1;
+    	}
+	
+	/* Copy the remaining elements of left_arr[], if there	are any */
+	while (i <= left_length)
+	{
+		arr(k) = left_arr(i);
+		i+=1;
+		k+=1;
+	}
+
+	/* Copy the remaining elements of right_arr[], if there	are any */
+	while (j <= right_length)
+	{
+		arr(k) = right_arr(j);
+		j+=1;
+		k+=1;
+	}
+}
+
+//parallel mergesort algorithm
+//can improve performance by adding heuristics -> When the array size becomes small such that creaing a task takes more time than compute time for sorting it, then dont spawn task, rather do it sequentially.
```

Comment:
Typo on "creating".  Also, might be nice to wrap these lines to 80 characters, it goes off the screen here.